code/create_term_vocab_from_gsheet.py

```diff
@@ -67,10 +68,38 @@ def fetch_gsheet_to_df(gsheet_id: str) -> pd.DataFrame:
     return vocab_df
 
 
+def remove_rows_with_invalid_reason(vocab_table: pd.DataFrame) -> pd.DataFrame:
+    """
+    Remove rows from the vocabulary table that have a non-empty 'invalid_reason' field,
+    i.e. terms that have been marked as invalid for some reason,
+    and remove the 'invalid_reason' column from the resulting filtered table.
+    """
+    if "invalid_reason" not in vocab_table.columns:
+        return vocab_table
+
+    invalid_term_mask = vocab_table["invalid_reason"].notna() & (
+        vocab_table["invalid_reason"].str.strip() != ""
+    )
+    if invalid_term_mask.any():
+        invalid_rows = vocab_table.loc[
+            invalid_term_mask, ["id", "name", "invalid_reason"]
+        ]
+        num_invalid_terms = len(invalid_rows)
+        print(f"{num_invalid_terms} invalid term(s) removed:")
+        for _, row in invalid_rows.iterrows():
+            print(
+                f"- id={row['id']}, name={row['name']}, invalid_reason={row['invalid_reason']}"
+            )
```

Comment:
🍒 want to add a logger for this? Maybe overkill given that we only run this in the GH CI. But could eventually store the logfiles in artefacts for later checking